core/src/main/java/tc/oc/pgm/stats/StatType.java

```diff
@@ -1,43 +1,87 @@
 package tc.oc.pgm.stats;
 
 import static net.kyori.adventure.text.Component.translatable;
-import static net.kyori.adventure.text.format.NamedTextColor.*;
+import static net.kyori.adventure.text.format.NamedTextColor.GREEN;
+import static net.kyori.adventure.text.format.NamedTextColor.RED;
+import static net.kyori.adventure.text.format.NamedTextColor.YELLOW;
 import static tc.oc.pgm.stats.StatsMatchModule.damageComponent;
 import static tc.oc.pgm.util.text.NumberComponent.number;
 
 import java.util.Locale;
 import net.kyori.adventure.text.Component;
+import tc.oc.pgm.util.math.Formula;
 
-public enum StatType {
-  KILLS,
-  DEATHS,
-  ASSISTS,
-  KILL_STREAK,
-  BEST_KILL_STREAK,
-  KILL_DEATH_RATIO,
-  LONGEST_BOW_SHOT {
-    private final String blocks = key + ".blocks";
+public sealed interface StatType {
+  Component makeNumber(Number number);
+
+  Component buildMessage(Component valueComponent);
+
+  Component component(StatHolder stats);
+
+  enum BaseStatType implements StatType {
+    KILLS,
+    DEATHS,
+    ASSISTS,
+    KILL_STREAK,
+    BEST_KILL_STREAK,
+    KILL_DEATH_RATIO,
+    LONGEST_BOW_SHOT {
+      private final String blocks = key + ".blocks";
+
+      @Override
+      public Component makeNumber(Number number) {
+        return translatable(blocks, number(number, YELLOW));
+      }
+    },
+    DAMAGE {
+      @Override
+      public Component makeNumber(Number number) {
+        return damageComponent(number.doubleValue(), GREEN);
+      }
+    };
+
+    public final String key = "match.stats.type." + name().toLowerCase(Locale.ROOT);
 
     @Override
     public Component makeNumber(Number number) {
-      return translatable(blocks, number(number, YELLOW));
+      return number(number, this == DEATHS ? RED : GREEN);
+    }
+
+    @Override
+    public Component component(StatHolder stats) {
+      return translatable(key, makeNumber(stats.getStat(this)));
     }
-  },
-  DAMAGE {
+
     @Override
-    public Component makeNumber(Number number) {
-      return damageComponent(number.doubleValue(), GREEN);
+    public Component buildMessage(Component valueComponent) {
+      return translatable(key, valueComponent);
     }
-  };
+  }
 
-  public final String key = "match.stats.type." + name().toLowerCase(Locale.ROOT);
-  public static final StatType[] VALUES = values();
+  final class FormulaStatType implements StatType {
```

Comment:
```suggestion
  final class OfFormula implements StatType {
```
Similar as above, would simplify to `StatType.OfFormula`

Reply:
Sure